src/context/ActionCreators.ts

```diff
@@ -0,0 +1,33 @@
+import { AuthFormData } from "src/common/interface";
+import { Dispatch, Types } from "src/context/context.interface";
+import { signUp, signIn, signOut } from "src/modules/AuthService";
+
+export const register = async (dispatch: Dispatch, payload: AuthFormData) => {
+  try {
+    dispatch({ type: Types.REQUEST });
+    await signUp(payload);
+  } catch (error) {
```

Comment:
Please do appropriate error handling, we should be able to know if the call fails and show an error.

Reply:
i was thinking about adding react-toastify for now to show error and success toast

Reply:
Actions should only call `dispatch` which modifies the `state`.
The component should be responsible for rendering the toast based on the current `state`.